README.md

```diff
@@ -745,6 +745,138 @@ or file contents that might later make it a lot easier to figure out what went w
 All of this verbose output is actually available all the time.  It’s stored in a log file.  After your (verbose or
 non-verbose) `etest` run, look at `etest.log`.
 
+## Mocking
+
+Ebash supports an extensive mocking framework called `emock`. It integrates well into the rest of ebash and etest in
+particular to make it very easy to mock out real system binaries and replace them with mock instances which do something
+else. It is very flexible in the behavior of the mocked function utilizing the many provided option flags.
+
+The simplest invocation of emock is to use the default of function mocking. In this mode, you simply supply the name
+of the binary you wish to mock, such as:
+
+    # emock "dmidecode"
+
+This will create and export a new function called 'dmidecode' that can be invoked instead of the real 'dmidecode'
+binary at '/usr/bin/dmidecode'. This also creates a function named 'dmidecode_real' which can be invoked to get access
+to the real underlying dmidecod binary at "/usr/sbin/dmidecode".
```

Comment:
Typo around dmidecode, but also, holy crap that's awesome to mock out the binary+path.